source/common/http/filter/ratelimit.cc

```diff
@@ -89,23 +89,32 @@ FilterHeadersStatus Filter::decodeHeaders(HeaderMap& headers, bool) {
   }
 
   const Router::RouteEntry* route = callbacks_->routeTable().routeForRequest(headers);
-  if (route && route->rateLimitPolicy().doGlobalLimiting()) {
-    std::vector<::RateLimit::Descriptor> descriptors;
-    for (const ActionPtr& action : config_->actions()) {
-      action->populateDescriptors(*route, descriptors, *config_, headers, *callbacks_);
+  if (route) {
```

Comment:
Can we do this in the doGlobalLimiting() block below. We don't need to do the runtime check if there is no way we are going to call.